pkg/apis/core/validation/shoot.go

```diff
@@ -1065,16 +1065,22 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 	var (
 		allErrs = field.ErrorList{}
 
-		workerNames                        = make(map[string]bool)
-		atLeastOneActivePool               = false
-		atLeastOnePoolWithCompatibleTaints = len(workers) == 0
+		workerNames                               = make(map[string]bool)
+		atLeastOneActivePool                      = false
+		atLeastOnePoolWithCompatibleTaints        = len(workers) == 0
+		atLeastOnePoolWithAllowedSystemComponents = false
 	)
 
 	for i, worker := range workers {
 		if worker.Minimum != 0 && worker.Maximum != 0 {
 			atLeastOneActivePool = true
 		}
 
+		// ToDo: Should this also validate that the pool allowing System components is not scaled to zero?
```

Comment:
Yes, the maximum must be >=1.
As far as I know, scale-to-zero is not implemented for all providers [1], so from my point of view the check should for `minimum>=1`.

[1] https://github.com/gardener/autoscaler/issues/27

Reply:
Done